static/js/filter.js

```diff
@@ -99,6 +99,10 @@ function message_matches_search_term(message, operator, operand) {
     case 'sender':
         return people.id_matches_email_operand(message.sender_id, operand);
 
+    case 'huddle-with':
+        // this is all handled server side
+        return true;
```

Comment:
I think you want to do something more similar to the `pm-with` logic here, checking is the user is a recipient.